protocol/messenger.go

```diff
@@ -377,6 +389,131 @@ func (m *Messenger) handleSendContactCode() error {
 	return err
 }
 
+// contactCodeAdvertisement attaches a protobuf.ChatIdentity to the given protobuf.ContactCodeAdvertisement,
+// if the `shouldPublish` conditions are met
+func (m *Messenger) handleContactCodeChatIdentity(cca *protobuf.ContactCodeAdvertisement) error {
+	contactCodeTopic := transport.ContactCodeTopic(&m.identity.PublicKey)
+	shouldPublish, err := m.shouldPublishChatIdentity(contactCodeTopic)
+	if err != nil {
+		return err
+	}
+
+	if shouldPublish {
+		cca.ChatIdentity, err = m.createChatIdentity("private-chat")
+		if err != nil {
+			return err
+		}
+
+		err = m.persistence.SaveWhenChatIdentityLastPublished(contactCodeTopic)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// handleStandaloneChatIdentity sends a standalone ChatIdentity message to a public channel if the publish criteria is met
+func (m *Messenger) handleStandaloneChatIdentity(chat *Chat) error {
+	if chat.ChatType != ChatTypePublic {
+		return nil
+	}
+	shouldPublishChatIdentity, err := m.shouldPublishChatIdentity(chat.ID)
+	if err != nil {
+		return err
+	}
+	if !shouldPublishChatIdentity {
+		return nil
+	}
+
+	ci, err := m.createChatIdentity("public-chat")
```

Comment:
maybe we could use const/enums for `public-chat` and `private-chat`?

Reply:
Yeah, that's good point. This actually went through my mind at the time of writing the code, then I added it to a mental list of todos that got forgot `shame_emoji`